scripts/main/album.js

```diff
@@ -1535,9 +1535,24 @@ album.apply_nsfw_filter = function () {
  *  - the method returns `true` for regular albums if and only if the album is
  *    owned by the currently authenticated user.
  *
+ * Note, for the time being this method contains a work-around in case
+ * no album is loaded, but the root view is visible.
+ * Currently, this is necessary, because this method is (erroneously) called
+ * for the root view as well.
+ * In order to determine whether the work-around for the root view needs to
+ * be applied, this method checks if the root view is visible based on the
+ * visibility of the corresponding headers.
+ * Hence, the caller must ensure that the appropriate header is set first
+ * in order to obtain a correct result from this method.
+ *
  * @returns {boolean}
  */
 album.isUploadable = function () {
+	// Work-around in case this method is called for the root view
+	if (visible.albums()) {
+		return lychee.rights.is_admin || (lychee.user !== null && !lychee.publicMode && lychee.rights.may_upload);
```

Comment:
```suggestion
		return lychee.rights.is_admin || lychee.rights.may_upload;
```

Why is the following not sufficient?

Reply:
I don't know.

I guess in the long run your suggestion is how it _should_ be, but this condition is the old condition which has been around forever. IMHO, in the long run `lychee.publicMode` should be nuked completely as this should be equivalent to `lychee.user !== null`.

The point is `lychee.user` is rather new and has only recently introduced by https://github.com/LycheeOrg/Lychee-front/pull/310 while `lychee.publicMode` has been there forever since the dawn of Lychee to distinguish between an unauthenticated user and the authenticated user. This goes back to the time when the only authenticated user was the admin, i.e. when Lychee was single-user.

I don't know which legacy code somewhere somehow sets `lychee.publicMode` for some weird reason to achieve a certain effect. Hence, I would like to keep it (for now) until all code which _sets_ `lychee.publicMode` is gone and we are sure that `lychee.publicMode` is never `true`.